examples/examples.rs

```diff
@@ -82,37 +80,26 @@ fn examples() -> Result<()> {
     // This API handles variadics using Heterogeneous Lists.  This is one way to
     // call a function with multiple parameters:
 
-    print.call::<_, ()>(
-        hlist!["hello", "again", "from", "rust"],
-    )?;
+    print.call::<_, ()>(("hello", "again", "from", "rust"))?;
 
     // You can bind rust functions to lua as well
 
-    let check_equal = lua.create_function(|lua, args| {
-        // Functions wrapped in lua receive their arguments packed together as
-        // MultiValue.  The first thing that most wrapped functions will do
-        // is "unpack" this MultiValue into its parts.  Due to lifetime type
-        // signature limitations, this cannot be done automatically from the
-        // function signature, but this will be fixed with ATCs.  Notice the use
-        // of the hlist macros again.
-        let hlist_pat![list1, list2] = lua.unpack::<HList![Vec<String>, Vec<String>]>(args)?;
-
+    let check_equal = lua.create_function(|_, (list1, list2): (Vec<String>, Vec<String>)| {
```

Comment:
Okay, I think I'm starting to follow here, sorry I'm still pretty noobish at these very complex lifetime problems.

> My experiments indicate that a closure borrowing the outside lua will not compile. In particular, this works, but not this, and I don't see a way to make the second version work. Therefore, the parameter needs to be kept.

Just to be clear, the only reason the first one (https://play.rust-lang.org/?gist=d6474ac18c718bdf037e0f979c3d7c5e&version=stable) works is because it's a self borrow, correct?  The compiler wouldn't actually LET you return that c or move that c at all, so that wouldn't be especially useful right?